modules/auxiliary/gather/ldap_esc_vulnerable_cert_finder.rb

```diff
@@ -648,7 +875,20 @@ def print_vulnerable_cert_info
       print_status("  Distinguished Name: #{hash[:dn]}")
       print_status("  Manager Approval: #{hash[:manager_approval] ? '%redRequired' : '%grnDisabled'}%clr")
       print_status("  Required Signatures: #{hash[:required_signatures] == 0 ? '%grn0' : '%red' + hash[:required_signatures].to_s}%clr")
-      print_good("  Vulnerable to: #{techniques.join(', ')}")
+      registry_object = @ldap_objects.find { |obj| obj[:type] == :registry_values }
+
+      if registry_object && registry_object[:values].present?
+        print_good("  Vulnerable to: #{techniques.join(', ')}")
+      else
+        print_good("  Vulnerable to: #{(techniques - %w[ESC9 ESC10]).join(', ')}")
+        if techniques.include?('ESC9')
+          print_warning('  Potentially vulnerable to: ESC9 (the template is in a vulnerable configuration but in order to exploit registry key StrongCertificateBindingEnforcement must not be set to 2)')
+        end
+        if techniques.include?('ESC10')
+          print_warning('  Potentially vulnerable to: ESC10 (the template is in a vulnerable configuration but in order to exploit registry key StrongCertificateBindingEnforcement must be set to 0 or CertificateMappingMethods must be set to 0x4)')
```

Comment:
It probably makes sense to just consistently use decimal notation for all the numbers here.

```suggestion
          print_warning('  Potentially vulnerable to: ESC10 (the template is in a vulnerable configuration but in order to exploit registry key StrongCertificateBindingEnforcement must be set to 0 or CertificateMappingMethods must be set to 4)')
```